File: internal/model/ssh.go
package model

type SSHConnectionProfile struct {
	ID             string `gorm:"column:id;primaryKey" json:"id"`
	Name           string `gorm:"column:name;not null" json:"name"`
	Host           string `gorm:"column:host;not null" json:"host"`
	Port           int    `gorm:"column:port;not null" json:"port"`
	User           string `gorm:"column:user;not null" json:"user"`
	AuthMethod     string `gorm:"column:auth_method;not null" json:"auth_method"`
	IdentityFile   string `gorm:"column:identity_file" json:"identity_file,omitempty"`
	ConnectTimeout int    `gorm:"column:connect_timeout;not null" json:"connect_timeout"`
	CreatedAt      int64  `gorm:"column:created_at;not null;index" json:"created_at"`
	UpdatedAt      int64  `gorm:"column:updated_at;not null;index" json:"updated_at"`
}

func (SSHConnectionProfile) TableName() string {
	return "ssh_connection_profiles"
}

type SSHKnownHost struct {
	Endpoint    string `gorm:"column:endpoint;primaryKey" json:"endpoint"`
	Host        string `gorm:"column:host;not null" json:"host"`
	Port        int    `gorm:"column:port;not null" json:"port"`
	KeyType     string `gorm:"column:key_type;not null" json:"key_type"`
	PublicKey   string `gorm:"column:public_key;type:text;not null" json:"-"`
	Fingerprint string `gorm:"column:fingerprint;not null" json:"fingerprint"`
	CreatedAt   int64  `gorm:"column:created_at;not null" json:"created_at"`
	UpdatedAt   int64  `gorm:"column:updated_at;not null" json:"updated_at"`
}

func (SSHKnownHost) TableName() string {
	return "ssh_known_hosts"
}

File: ui/src/api/ssh.ts
import { request } from "@/api/request";

export type SSHAuthMethod = "auto" | "agent" | "private_key" | "password";

export interface SSHAuthSecrets {
  password?: string;
  private_key?: string;
  passphrase?: string;
}

export interface SSHProfile {
  id: string;
  name: string;
  host: string;
  port: number;
  user: string;
  auth_method: SSHAuthMethod;
  identity_file?: string;
  connect_timeout: number;
  connected: boolean;
  created_at: number;
  updated_at: number;
}

export interface SSHProfileInput {
  name?: string;
  host: string;
  port?: number;
  user?: string;
  auth_method?: SSHAuthMethod;
  identity_file?: string;
  connect_timeout?: number;
}

export interface SSHHostKeyChallenge {
  id: string;
  profile_id: string;
  endpoint: string;
  key_type: string;
  fingerprint: string;
  expires_at: number;
}

export const sshApi = {
  listProfiles: () => request<{ profiles: SSHProfile[] }>("/ssh/profiles"),

  createProfile: (input: SSHProfileInput) =>
    request<{ profile: SSHProfile }>("/ssh/profiles", {
      method: "POST",
      body: JSON.stringify(input),
    }),

  updateProfile: (id: string, input: Partial<SSHProfileInput>) =>
    request<{ profile: SSHProfile }>(`/ssh/profiles/${encodeURIComponent(id)}`, {
      method: "PATCH",
      body: JSON.stringify(input),
    }),

  deleteProfile: (id: string) =>
    request<{ ok: boolean }>(`/ssh/profiles/${encodeURIComponent(id)}`, {
      method: "DELETE",
    }),

  connect: (id: string, auth: SSHAuthSecrets = {}) =>
    request<{ ok: boolean; profile_id: string; connected: boolean }>(
      `/ssh/profiles/${encodeURIComponent(id)}/connect`,
      {
        method: "POST",
        body: JSON.stringify({ auth }),
      }
    ),

  disconnect: (id: string) =>
    request<{ ok: boolean; profile_id: string; connected: boolean }>(
      `/ssh/profiles/${encodeURIComponent(id)}/disconnect`,
      { method: "POST" }
    ),

  confirmHostKey: (challengeId: string) =>
    request<{ ok: boolean; endpoint: string; key_type: string; fingerprint: string }>(
      `/ssh/host-key-challenges/${encodeURIComponent(challengeId)}/confirm`,
      { method: "POST" }
    ),

  resetKnownHost: (profileId: string, expectedFingerprint: string) =>
    request<{ ok: boolean; endpoint: string; fingerprint: string; connected: boolean }>(
      `/ssh/profiles/${encodeURIComponent(profileId)}/known-host`,
      {
        method: "DELETE",
        body: JSON.stringify({ expected_fingerprint: expectedFingerprint }),
      }
    ),
};
